cloud.foundry.cli/src/main/java/cloud/foundry/cli/operations/ApplicationsOperations.java

```diff
@@ -169,109 +296,256 @@ public ApplicationsOperations(DefaultCloudFoundryOperations cloudFoundryOperatio
+    /**
+     * Prepares a request for binding an app to a service.
+     * The resulting mono is preconfigured such that it will perform logging.
+     *
+     * @param applicationName the app that should be bound to the service
+     * @param serviceName the service to which the app should be bound
+     * @return mono which can be subscribed on to trigger the app binding
+     * @throws NullPointerException if any of the arguments is null
+     */
+    public Mono<Void> bindToService(String applicationName, String serviceName) {
+        checkNotNull(applicationName);
+        checkNotNull(serviceName);
+
+        BindServiceInstanceRequest bindServiceRequest = BindServiceInstanceRequest.builder()
+                .applicationName(applicationName)
+                .serviceInstanceName(serviceName)
+                .build();
+
+        return cloudFoundryOperations.services().bind(bindServiceRequest)
+                .doOnSuccess(aVoid -> {
+                    log.debug("Bind app:", applicationName);
+                    log.debug("To service:", serviceName); })
+                .doOnSuccess(aVoid -> log.info("Bound the app", applicationName,
+                        "to the service", serviceName));
+    }
+
+    /**
+     * Prepares a request for unbinding an app from a service.
+     * The resulting mono is preconfigured such that it will perform logging.
+     *
+     * @param applicationName the app that should be unbound from the service
+     * @param serviceName the service from which the app should be unbound
+     * @return mono which can be subscribed on to trigger the app unbinding
+     * @throws NullPointerException if any of the arguments is null
+     */
+    public Mono<Void> unbindFromService(String applicationName, String serviceName) {
+        checkNotNull(applicationName);
+        checkNotNull(serviceName);
+
+        UnbindServiceInstanceRequest unbindServiceRequest = UnbindServiceInstanceRequest.builder()
+                .applicationName(applicationName)
+                .serviceInstanceName(serviceName)
+                .build();
+
+        return cloudFoundryOperations.services().unbind(unbindServiceRequest)
+                .doOnSuccess(aVoid -> {
+                    log.debug("Unbind app:", applicationName);
+                    log.debug("From service:", serviceName); })
+                .doOnSuccess(aVoid -> log.info("Unbound the app", applicationName,
+                        "from the service", serviceName));
     }
 
-    private Mono<ApplicationDetail> getApplicationDetail(String appName) {
+    /**
+     * Prepares a request for adding a route to an app.
+     * The resulting mono is preconfigured such that it will perform logging.
+     *
+     * @param applicationName the app to which the route should be added
+     * @param route the domain of the route to add to the app
+     * @return mono which can be subscribed on to trigger the route addition, that delivers the port number of the route
```

Comment:
The port number is no longer delivered
```suggestion
     * @return mono which can be subscribed on to trigger the route addition
```